saliency_metrics/metrics/insertion_deletion/insertion_deletion_metric.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Dict, List
+
+import numpy as np
+import torch
+from scipy.integrate import trapezoid
+from torchvision.transforms import GaussianBlur
+
+from saliency_metrics.metrics.build_metric import ReInferenceMetric
+from saliency_metrics.models.build_classifier import build_classifier
+from .insertion_deletion_perturbation import ProgressivePerturbation
+from .insertion_deletion_result import InsertionDeletionResult
+
+
+class InsertionDeletion(ReInferenceMetric):
+    def __init__(
+        self,
+        classifier_cfg: Dict,
+        forward_batch_size: int = 128,
+        perturb_step_size: int = 10,
+        sigma: float = 5.0,
+        summarized: bool = False,
+    ) -> None:
+
+        self._result = InsertionDeletionResult(summarized)
+
+        self.classifier = build_classifier(classifier_cfg)
+        # TODO - check
+        # self.classifier.eval()
+        self.gaussian_blur = GaussianBlur(int(2 * sigma - 1), sigma)
+        self.forward_batch_size = forward_batch_size
+        self.perturb_step_size = perturb_step_size
+
+    # TODO - check default value for img_path
+    def evaluate(self, img: torch.Tensor, smap: torch.Tensor, target: int, **kwargs: Any) -> Dict:
+        num_pixels = torch.numel(smap)
+        # TODO - check
+        if self.perturb_step_size <= 0 or self.perturb_step_size >= num_pixels:
```

Comment:
It should be checked in the init method but not here.

Reply:
The check is dependent on num_pixels which could change with every evaluate call, hence I decided to place it in evaluate. I could break this into two checks - check if perturb_step_size and forward_batch_size are greater than zero in init method and check if perturb_step_size<num_pixels in evaluate method.